src/averageAges.js

```diff
@@ -37,7 +38,13 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredPeople = !withChildren
+    ? people.filter(el => (el.sex !== 'm'))
+    : people.filter(el => (people.find(child => el.name === child.mother)
+    && (el.sex !== 'm')));
```

Comment:
```suggestion
    : people.filter(el => (people.find(child => el.name === child.mother)));
```
You could delete this redundant condition.